specs/selectors/fixtures/selector-fixtures-1.md

```diff
@@ -338,6 +338,79 @@ if there's several things you want in a row.
 
 ---
 
+### slice matching
+
+Matching a node that is a string or bytes can match a subset of the string or
+bytes. This is primarily helpful in the case of an ADL that reifies a complex
+representation into a simplified single string or bytes node, such as one that
+represents a multi-block bytes node as a single bytes node.
+
+#### data
+
+[testmark]:# (match-subset/data)
+```json
+"a long string that we want to match a subset of"
+```
+
+#### selector
+
+[testmark]:# (match-subset/selector)
+```json
+{
+	".": {
+		"subset": {
+			"[": 30,
+			"]": 44
+		}
+	}
+}
+```
+
+#### expected visit events
+
+[testmark]:# (match-subset/expect-visit)
+```ipldsch
+{"path": "", "node": {"string": "match a subset"}, "matched": true}
+```
+
+---
+
+### slice matching with negative indexes and extreme values
+
+Negative indexes for both "from" and "to" can be used to count from the end of
+the string or bytes node. The "to" index can be greater than the length of the
+string or bytes node, in which case it is treated as the length of the string
+or bytes node.
+
+#### data
+
+[testmark]:# (match-subset-extremities/data)
+```json
+"a long string that we want to match a subset of"
+```
+
+#### selector
+
+[testmark]:# (match-subset-extremities/selector)
+```json
+{
+	".": {
+		"subset": {
+			"[": -9,
+			"]": 9223372036854775807
```

Comment:
Added more comments about it. For the record on this topic so it doesn't get lost, I said this on Slack:

Regarding the "how to select the last byte" problem, I've considered 3 options, currently going with the first:
* Just allow overflow on the To value so you can use MaxInt64 or other large number as your end signal. I put that in the schema spec and it's used in the tests. -1,9999999... selects last byte.  A bit ugly but not schema parse breaking.
* Making the To value optional or nullable, so that can be an "end" signal and -1,null as a range would be the last byte. But this is breaking and it'd be a schema parse error for existing schemas.
* Make To inclusive, like in the http range spec and Trustless Gateway. I suspect this might be why they made the range [From,To] rather than the more typical [From,To), because negative indexes get annoying. This would be a breaking change to existing behaviour, maybe not terrible kind of breaking though.

Reply:
i think that's reasonable.